app/models/concerns/custom_field_addable.rb

```diff
@@ -118,6 +124,12 @@ def custom_field?(field_identifier)
     self.class.custom_field?(field_identifier, model: self)
   end
 
+  def unanswered?(field_identifier)
+    value = custom_value(field_identifier)
+    field = custom_field(field_identifier)
+    value.blank? && field.descendants.all? { |i| custom_value(i.id).blank? }
+  end
```

Comment:
This seems like a hack to me and a potential performance nightmare. Why can't we call `unanswered?` with no arguments on the appropriate object? The answer is likely that the system is not properly built out to handle that, but I don't think we should just keep adding hacks like this. We need to build it out properly. This is what I was planning to do as part of the progress bar issue. Thoughts?